GTI Release 2.0.0

crestdatasystems opened the pull request:
<!-- REMINDER: THIS IS A PUBLIC REPO DO NOT POST HERE SECRETS/SENSITIVE DATA -->
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Description

### Added following Integrations:

#### Google Threat Intelligence - ASM Issues

-  This integration allows the creation of incidents based on ASM Issues from Google Threat Intelligence.

- Added the following commands:

	- ***gti-asm-issue-list***
	- ***gti-asm-issue-get***
	- ***gti-asm-issue-status-update***

#### Google Threat Intelligence - DTM Alerts

- This integration allows the creation of incidents based on DTM Alerts from Google Threat Intelligence.

- Added the following commands:

	- ***gti-dtm-alert-list***
	- ***gti-dtm-alert-get***
	- ***gti-dtm-alert-status-update***

## Must have
- [x] Tests
- [x] Documentation

merit-maita commented on Packs/GoogleThreatIntelligence/ReleaseNotes/2_0_0.md:
```diff
@@ -0,0 +1,163 @@
+#### Integrations
+
+##### New: Google Threat Intelligence - ASM Issues
```
The heading format for new content items should not include the "New:" prefix. Please remove it from this line and all similar headings throughout the file.

merit-maita commented on Packs/GoogleThreatIntelligence/ReleaseNotes/2_0_0.md:
```diff
@@ -0,0 +1,163 @@
+
+##### New: Google Threat Intelligence - ASM Issues
+
+- This integration allows the creation of incidents based on ASM Issues from Google Threat Intelligence.
```
The release note description does not start with an approved template phrase. Please rephrase this line and line 15 to use one of the standard prefixes.

merit-maita commented on Packs/GoogleThreatIntelligence/Integrations/GoogleThreatIntelligenceASMIssues/GoogleThreatIntelligenceASMIssues.yml:
The output type is set to `Unknown`. Please change it to `Number` for consistency.

merit-maita commented on Packs/GoogleThreatIntelligence/Integrations/GoogleThreatIntelligenceASMIssues/GoogleThreatIntelligenceASMIssues.yml:
The output type is set to `Unknown`. Please change it to `Number` for consistency.

merit-maita commented on Packs/GoogleThreatIntelligence/Integrations/GoogleThreatIntelligenceASMIssues/GoogleThreatIntelligenceASMIssues.yml:
Invalid output type 'list'. Please change the type to `List`.

merit-maita commented on Packs/GoogleThreatIntelligence/Integrations/GoogleThreatIntelligenceDTMAlerts/GoogleThreatIntelligenceDTMAlerts.yml:
```diff
@@ -0,0 +1,946 @@
+      type: String
+    - contextPath: GoogleThreatIntelligenceDTMAlerts.Alerts.alerts.tags
+      description: Tags associated with the alert.
+      type: Unknown
```
The output type is set to `Unknown`. Please change the type to `List`.

merit-maita commented on Packs/GoogleThreatIntelligence/Integrations/GoogleThreatIntelligenceDTMAlerts/GoogleThreatIntelligenceDTMAlerts.yml:
```diff
@@ -0,0 +1,946 @@
+      type: Number
+    - contextPath: GoogleThreatIntelligenceDTMAlerts.Alerts.alerts.confidence_reasoning.confidence_count
+      description: Total number of samples considered for confidence calculation.
+      type: Unknown
```
The output type is set to `Unknown`. Please change the type to `Number`.

merit-maita commented on Packs/GoogleThreatIntelligence/Integrations/GoogleThreatIntelligenceDTMAlerts/GoogleThreatIntelligenceDTMAlerts.yml:
```diff
@@ -0,0 +1,946 @@
+      type: Unknown
+    - contextPath: GoogleThreatIntelligenceDTMAlerts.Alerts.alerts.confidence_reasoning.explanation
+      description: Explanation behind the confidence score.
+      type: Unknown
```
The output type is set to `Unknown`. Please change the type to `String`.

merit-maita commented on Packs/GoogleThreatIntelligence/Integrations/GoogleThreatIntelligenceASMIssues/GoogleThreatIntelligenceASMIssues_description.md:
```diff
@@ -0,0 +1,22 @@
+### Authorization:
```
- The integration description is missing a main title and an introductory section, and it uses incorrect heading levels. Please add a level 2 header with the integration name, a brief paragraph explaining the integration's purpose, and update the existing section headers from level 3 (`###`) to level 2 (`##`).
- The documentation is missing the required 'Permissions' and 'Troubleshooting' sections. Please add these sections to provide more complete guidance.

merit-maita commented on Packs/GoogleThreatIntelligence/Integrations/GoogleThreatIntelligenceASMIssues/GoogleThreatIntelligenceASMIssues_description.md:
```diff
@@ -0,0 +1,22 @@
+
+### Instance Configuration
+
+1. Configure an Google Threat Intelligence - ASM Issues integration instance with valid API Key.
```
Please correct the grammatical error from "an Google" to "a Google".

merit-maita commented on Packs/GoogleThreatIntelligence/Integrations/GoogleThreatIntelligenceASMIssues/GoogleThreatIntelligenceASMIssues_description.md:
```diff
@@ -0,0 +1,22 @@
+
+1. Configure an Google Threat Intelligence - ASM Issues integration instance with valid API Key.
+2. Click **Test** to validate the connection.
+3. To fetch ASM Issues as an incidents in XSOAR, select the option `Fetches incidents` and follow the table to update configuration parameters.
```
Please correct the grammatical error from "as an incidents" to "as incidents".

merit-maita commented on Packs/GoogleThreatIntelligence/Integrations/GoogleThreatIntelligenceDTMAlerts/README.md:
```diff
@@ -0,0 +1,929 @@
+
+Your API key can be found in your GoogleThreatIntelligence account user menu, clicking on your avatar:
+
+![How to get api key in GoogleThreatIntelligence](../../doc_files/ddeb298-Screen_Shot_2019-10-17_at_3_17_04_PM.png)
```
The image referenced here for acquiring an API key appears to be from a different product, not Google Threat Intelligence. The filename `Screen_Shot_2019-10-17_at_3_17_04_PM.png` also suggests it might be outdated. Please replace this with a relevant image showing how to get an API key from the Google Threat Intelligence platform.

merit-maita commented on Packs/GoogleThreatIntelligence/Integrations/GoogleThreatIntelligenceDTMAlerts/README.md:
```diff
@@ -0,0 +1,929 @@
+>
+>|Alert ID|Title|Alert Summary|Alert Type|Severity|Status|Monitor ID|Monitor Name|Indicator Score|Created At|Updated At|Tags|
+>|---|---|---|---|---|---|---|---|---|---|---|---|
+>| dummy_0000 | Found topic "google" posted by actor "sevenbump" on channel "social" | This message advertises a service that provides fraudulent Google verification badges for any account. <br> | Message | Low | Read | dummy_monitor_id | Initial Access Broker | 11 | 2025-05-21T11:28:02.646Z | 2025-05-22T07:09:51.257Z | attempt,<br>Test |
```
The `<br>` tags in the human-readable output table (lines 475 and 476) are not self-closed. To ensure the documentation renders correctly on MDX-based platforms, please update them to be self-closing (`<br/>`).

merit-maita commented on Packs/GoogleThreatIntelligence/Classifiers/classifier-Google_Threat_Intelligence_DTM_Alerts-_Incoming_Mapper.json:
```diff
@@ -0,0 +1,357 @@
+                        "root": "created_at",
+                        "transformers": []
+                    }
+                },
```
The 'GTI DTM Alert Doc Description' field is mapped with a root of 'description'. However, in the provided test data, 'description' is nested under 'doc'. Please adjust the mapping to 'root: doc, accessor: description'.

merit-maita commented on Packs/GoogleThreatIntelligence/Classifiers/classifier-Google_Threat_Intelligence_DTM_Alerts-_Incoming_Mapper.json:
```diff
@@ -0,0 +1,357 @@
+                        "root": "doc",
+                        "transformers": []
+                    }
+                },
```
The 'GTI DTM Alert Monitor Name' and 'Tags' fields are present in `dtm_alerts_lists.json` but are missing from `dtm_alert_get.json`. Please ensure consistency in test data or clarify if these fields are conditionally present.

merit-maita reviewed (changes requested):
Hey, thanks for adding the new Google Threat Intelligence integration, Great work on this pack!

- [Minor Improvement] Consider adding a dark theme variant of this logo named `GoogleThreatIntelligenceDTMAlerts_image_dark.png` to ensure optimal display in XSOAR's dark mode.

crestdatasystems commented on Packs/GoogleThreatIntelligence/Integrations/GoogleThreatIntelligenceDTMAlerts/GoogleThreatIntelligenceDTMAlerts.yml:
```diff
@@ -0,0 +1,946 @@
+  required: false
+  section: Connect
+- display: Incident type
+  name: incidentType
```
The XSOAR configuration parameter like incident type are auto-generated. So, we have kept the naming same.
Also, the [examples from the content repo](https://github.com/demisto/content/blob/master/Packs/HelloWorld/Integrations/HelloWorld/HelloWorld.yml#L36C9-L36C21) follows the same format.

crestdatasystems commented:
Hi @Benimanela,

Thanks for reviewing the PR! We’ve implemented the suggested changes, and here’s a summary of the updates:

## General

All files have been formatted using `demisto-sdk` to ensure alignment with XSOAR standards.

## Mappers

- Added `first_seen` and `created_at` fields to map to the common field `Occurred` in the incoming mappers for both integrations (ASM Issues and DTM Alerts).
- The `severity` field continues to be mapped through the integration code.
- For `status`, we’ve introduced a custom incident field to display the external system’s status in the layout.

## Layout

- Severity is populated directly through the integration code.
- Status is managed via the custom incident field, ensuring both DTM Alerts and ASM Issues reflect their respective external statuses correctly.

### ASM Issue Layout

> <img width="1294" height="655" alt="GTI ASM Issue Layout" src="https://github.com/user-attachments/assets/c5f9eb2d-315c-43a1-a1d5-79471eaff1df" />

---

### DTM Alert Layout

> <img width="1435" height="719" alt="GTI DTM Alert Layout" src="https://github.com/user-attachments/assets/9092b032-9c2f-46c7-bca0-0de9178af7e7" />

## Issue Observed in the PR

- A default value has been set for the incident type parameter in both integrations.
- In XSOAR 6.14, everything works as expected. However, in XSOAR 8.9, DTM Alert incidents are being dropped when using the default incident type.
- The same configuration works correctly for ASM Issues, which are fetched successfully with their default type.
- Please share your thoughts on providing a default incident type. Additionally, could you confirm if there might be an issue with the incident type itself or with its definition in the integration YML?

merit-maita commented:
> Hi @merit-maita @Benimanela @DeanArbel,
> 
> It has been over two weeks since we resolved the only remaining incident-type issue. We haven’t received any further comments from the reviewers, and no additional actions have been taken.
> 
> We kindly request your assistance in progressing this PR, as the Google team is expecting it to be published on the marketplace as soon as possible.

hi @crestdatasystems 
sorry for the delay in handling the contribution, im on it today
